algorithms/src/std_algorithms/impl/Kokkos_RandomAccessIterator.hpp

```diff
@@ -111,77 +118,87 @@ class RandomAccessIterator< ::Kokkos::View<DataType, Args...> > {
 
   KOKKOS_FUNCTION
   reference operator[](difference_type n) const {
-    return m_view(m_current_index + n);
+    if constexpr (is_always_contiguous)
+      return *(m_data + n);
+    else
+      return *(m_data + n * m_stride);
   }
 
   KOKKOS_FUNCTION
   iterator_type& operator+=(difference_type n) {
-    m_current_index += n;
+    if constexpr (is_always_contiguous)
+      m_data += n;
+    else
+      m_data += n * m_stride;
     return *this;
   }
 
   KOKKOS_FUNCTION
   iterator_type& operator-=(difference_type n) {
-    m_current_index -= n;
+    if constexpr (is_always_contiguous)
+      m_data -= n;
+    else
+      m_data -= n * m_stride;
     return *this;
   }
 
   KOKKOS_FUNCTION
   iterator_type operator+(difference_type n) const {
-    return iterator_type(m_view, m_current_index + n);
+    auto it = *this;
+    it += n;
+    return it;
```

Comment:
It is not clear to me why you don't use `is_contiguous` here ?

Reply:
It is used underneath in `it += n;` statement.